qiskit_experiments/library/characterization/fine_frequency.py

```diff
@@ -0,0 +1,129 @@
+        The idle time is followed by a rz rotation with an angle :math:`n\pi/2` and a final
+        sx gate. If the frequency of the pulses match the frequency of the qubit then the
+        sequence :math:`n\in[0,1,2,3,4,...]` will result in the sequence of measured qubit
+        populations :math:`[1, 0.5, 0, 0.5, 1, ...]` due to the rz rotation. If the frequency
+        of the pulses do not match the qubit frequency then the qubit will precess in the
+        drive frame during the idle time and phase errors will build-up. By fitting the measured
+        points we can extract the error in the qubit frequency. The circuit that are run are
+
+        .. parsed-literal::
+
+                    ┌────┐┌────────────────┐┌──────────┐┌────┐ ░ ┌─┐
+                 q: ┤ √X ├┤ Delay(n*τ[dt]) ├┤ Rz(nπ/2) ├┤ √X ├─░─┤M├
+                    └────┘└────────────────┘└──────────┘└────┘ ░ └╥┘
+            meas: 1/══════════════════════════════════════════════╩═
+                                                                  0
+    # section: analysis_ref
+        :py:class:`FineFrequencyAnalysis`
+    """
+
+    def __init__(
+        self,
+        qubit: int,
+        delay_duration: int,
+        backend: Optional[Backend] = None,
+        repetitions: Optional[List[int]] = None,
+    ):
+        """Setup a fine frequency experiment on the given qubit.
+
+        Args:
+            qubit: The qubit on which to run the fine frequency characterization experiment.
+            delay_duration: The duration of the delay at :math:`n=1`.
+            backend: Optional, the backend to run the experiment on.
+            repetitions: The number of repetitions, if not given then the default value
+                from the experiment default options will be used.
+        """
+        super().__init__([qubit], analysis=FineFrequencyAnalysis(), backend=backend)
+
+        if repetitions is not None:
+            self.set_experiment_options(repetitions=repetitions)
+
+        self.set_experiment_options(delay_duration=delay_duration)
+
+    @classmethod
+    def _default_experiment_options(cls) -> Options:
+        r"""Default values for the fine frequency experiment.
+
+        Experiment Options:
+            repetitions (List[int]): A list of the number of times that the identity is repeated.
+            delay_duration (int): The duration of the delay as the number of arbitrary waveform
+                generator samples it contains. The total length of the delay will be n times
+                ``delay_duration`` where n also determines the rotation angle of the ``RZGate``
+                by :math:`n \pi/2`.
+        """
+        options = super()._default_experiment_options()
+        options.repetitions = list(range(40))
+        options.delay_duration = None
+
+        return options
+
+    @classmethod
+    def _default_transpile_options(cls) -> Options:
+        """Default transpiler options."""
+        options = super()._default_transpile_options()
```

Comment:
probably basis_gate?

Reply:
Yes, I think that is okay. See here: 7a90851. I tested the characterization version on hardware and it runs.